app/javascript/tests/patient/close_contact/CloseContactModal.test.js

```diff
@@ -0,0 +1,516 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Button, Form, Modal } from 'react-bootstrap';
+import ReactTooltip from 'react-tooltip';
+import _ from 'lodash';
+
+import CloseContactModal from '../../../components/patient/close_contacts/CloseContactModal';
+import DateInput from '../../../components/util/DateInput';
+import PhoneInput from '../../../components/util/PhoneInput';
+import { mockCloseContactBlank, mockCloseContact2 } from '../../mocks/mockCloseContacts';
+
+const onSaveMock = jest.fn();
+const onCloseMock = jest.fn();
+const ASSIGNED_USERS = [123234, 512678, 910132];
+const INPUT_FIELDS = [
+  { name: 'first_name', label: 'First Name', type: 'text' },
+  { name: 'last_name', label: 'Last Name', type: 'text' },
+  { name: 'primary_telephone', label: 'Phone Number', type: 'phone' },
+  { name: 'email', label: 'Email', type: 'text' },
+  { name: 'last_date_of_exposure', label: 'Last Date of Exposure', type: 'date' },
+  { name: 'assigned_user', label: 'Assigned User', type: 'datalist', options: ASSIGNED_USERS },
+  { name: 'notes', label: 'Notes', type: 'text' },
+];
+let activeCC;
+
+function getWrapper(editMode) {
+  activeCC = editMode ? mockCloseContact2 : mockCloseContactBlank;
+  return shallow(<CloseContactModal currentCloseContact={activeCC} editMode={editMode} onClose={onCloseMock} onSave={onSaveMock} assigned_users={ASSIGNED_USERS} />);
+}
+
+describe('CloseContactModal', () => {
+  it('Properly renders main components when adding new close contact', () => {
+    const wrapper = getWrapper(false);
+    expect(wrapper.find(Modal).exists()).toBe(true);
+    expect(wrapper.find(Modal).find('h1').text()).toEqual('Add New Close Contact');
+    expect(wrapper.find(Modal.Header).exists()).toBe(true);
+    expect(wrapper.find(Modal.Title).text()).toEqual('Add New Close Contact');
+    expect(wrapper.find(Modal.Body).exists()).toBe(true);
+    expect(wrapper.find(Modal.Body).find(Form.Group).length).toEqual(INPUT_FIELDS.length);
+    INPUT_FIELDS.forEach((field, fieldIndex) => {
+      const input = wrapper.find(Modal.Body).find(Form.Group).at(fieldIndex);
+      if (field.type === 'text') {
+        expect(input.find(Form.Label).exists()).toBe(true);
+        expect(input.find(Form.Label).text()).toContain(field.label);
```

Comment:
Could these two lines be extracted to below line 41 outside of the if statements since it looks like they are tested for each `field.type`?

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1184/commits/428cc44932d92387fd8c2689c5c6095b2619b624